jetty-http/src/main/java/org/eclipse/jetty/http/compression/NBitIntegerParser.java

```diff
@@ -11,12 +11,27 @@
 // ========================================================================
 //
 
-package org.eclipse.jetty.http3.qpack.internal.util;
+package org.eclipse.jetty.http.compression;
 
 import java.nio.ByteBuffer;
 
 public class NBitIntegerParser
 {
+    public static int decode(ByteBuffer buffer, int prefix) throws EncodingException
+    {
+        // TODO: This is a fix for HPACK as it already takes the first byte of the encoded integer.
+        if (prefix != 8)
+            buffer.position(buffer.position() - 1);
```

Comment:
Could this fix be put in a utility method on hpack/qpack?   Does it also apply to qpack?

Reply:
No this is specific to HPACK, it does something weird where it peeks at the first byte and then expects this to put it back.

Looks like I can also remove this static and do this in HPACK only.